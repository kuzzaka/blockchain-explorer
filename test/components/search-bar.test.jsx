import React from 'react';
import { mount, shallow } from 'enzyme';
import SearchBar from '../../components/search-bar';
import createStore from '../../redux/store';

const normalState = {
  query: 'hash',
  error: false,
};
const withErrorState = {
  query: 'none',
  error: { error: true },
};

test('SearchBar', () => {
  const store = createStore(normalState);
  const spy = jest.spyOn(store, 'dispatch');
  const wrapper = mount(<SearchBar store={store} />);
  expect(wrapper.find('input')).toExist();
  expect(wrapper.find('input')).toHaveValue('hash');
  wrapper.find('input').simulate('change');
  expect(spy).toHaveBeenCalled();
});

test('normal SearchBar snapshot', () => {
  const wrapper = shallow(<SearchBar store={createStore(normalState)} />);
  expect(wrapper).toMatchSnapshot();
});

test('with error SearchBar snapshot', () => {
  const wrapper = shallow(<SearchBar store={createStore(withErrorState)} />);
  expect(wrapper).toMatchSnapshot();
});
