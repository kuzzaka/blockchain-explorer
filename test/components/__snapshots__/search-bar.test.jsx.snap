// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normal SearchBar snapshot 1`] = `
<SearchBar
  error={false}
  query="hash"
  searchHash={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSagaTask": [Function],
      "sagaTask": Object {
        "@@redux-saga/TASK": true,
        "cancel": [Function],
        "cont": undefined,
        "done": Promise {},
        "error": [Function],
        "id": 9,
        "isAborted": [Function],
        "isCancelled": [Function],
        "isRunning": [Function],
        "joiners": Array [],
        "name": "rootSaga",
        "result": [Function],
        "setContext": [Function],
      },
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "runSagaTask": [Function],
        "sagaTask": Object {
          "@@redux-saga/TASK": true,
          "cancel": [Function],
          "cont": undefined,
          "done": Promise {},
          "error": [Function],
          "id": 9,
          "isAborted": [Function],
          "isCancelled": [Function],
          "isRunning": [Function],
          "joiners": Array [],
          "name": "rootSaga",
          "result": [Function],
          "setContext": [Function],
        },
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`with error SearchBar snapshot 1`] = `
<SearchBar
  error={
    Object {
      "error": true,
    }
  }
  query="none"
  searchHash={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSagaTask": [Function],
      "sagaTask": Object {
        "@@redux-saga/TASK": true,
        "cancel": [Function],
        "cont": undefined,
        "done": Promise {},
        "error": [Function],
        "id": 13,
        "isAborted": [Function],
        "isCancelled": [Function],
        "isRunning": [Function],
        "joiners": Array [],
        "name": "rootSaga",
        "result": [Function],
        "setContext": [Function],
      },
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "runSagaTask": [Function],
        "sagaTask": Object {
          "@@redux-saga/TASK": true,
          "cancel": [Function],
          "cont": undefined,
          "done": Promise {},
          "error": [Function],
          "id": 13,
          "isAborted": [Function],
          "isCancelled": [Function],
          "isRunning": [Function],
          "joiners": Array [],
          "name": "rootSaga",
          "result": [Function],
          "setContext": [Function],
        },
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
